juju/juju
# Backport backups facade v3 support

Add support for the juju 2.9 client to use the backups v3 facade which is what juju 3controllers implement. The only difference is that create() does not return the controller cacert and key in the metadata - these are unused anyway.

Also, remove the client side "charmhub" facade version check - this was used to see if we default to ch or cs. There's no aother client side charmhub api used, so instead check for "resources" version - it was "1" in juju 2.8 and "2" or greater in 2.9 onwards.

Now the client facade versions used by juju 2.9 and 3.0 are the same and the 2.9 CLI can talk to a 3.0 controller.

## QA steps

bootstrap a 3.0 controller
use this client to do a backup
use this client to deploy a ch charm without using the "ch" prefix, check the origin after to make sure it's charmhub.
